systemtest/src/test/java/io/strimzi/systemtest/upgrade/OlmUpgradeIsolatedST.java

```diff
@@ -55,117 +54,121 @@ public class OlmUpgradeIsolatedST extends AbstractUpgradeST {
-        OlmUtils.waitUntilNonUsedInstallPlanIsPresent(INFRA_NAMESPACE, toVersion);
 
-        // Cluster Operator
-        clusterOperator.getOlmResource().upgradeClusterOperator();
-
-        // wait until RU is finished
-        zkPods = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(clusterOperator.getDeploymentNamespace(), zkSelector, 3, zkPods);
-        kafkaPods = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(clusterOperator.getDeploymentNamespace(), kafkaSelector, 3, kafkaPods);
-        eoPods = DeploymentUtils.waitTillDepHasRolled(clusterOperator.getDeploymentNamespace(), KafkaResources.entityOperatorDeploymentName(clusterName), 1, eoPods);
-        // ======== Cluster Operator upgrade ends ========
+        // Cluster Operator upgrade
+        clusterOperator.getOlmResource().upgradeClusterOperator(clusterOperator.getDeploymentNamespace(), upgradeOlmConfig.getCsvName());
 
         clusterOperatorDeploymentName = ResourceManager.kubeClient().getDeploymentNameByPrefix(Environment.OLM_OPERATOR_DEPLOYMENT_NAME);
         LOGGER.info("New deployment name of Cluster Operator is {}", clusterOperatorDeploymentName);
         ResourceManager.setCoDeploymentName(clusterOperatorDeploymentName);
 
-        // verification that cluster operator has correct version (install-plan) - strimzi-cluster-operator.v[version]
+        // Verification that Cluster Operator has been upgraded to a correct version
         String afterUpgradeVersionOfCo = kubeClient().getCsvWithPrefix(clusterOperator.getDeploymentNamespace(), upgradeOlmConfig.getOlmAppBundlePrefix()).getSpec().getVersion();
-
-        // if HEAD -> 6.6.6 version
         assertThat(afterUpgradeVersionOfCo, is(toVersion));
 
+        // Wait for Rolling Update to finish
+        zkPods = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(clusterOperator.getDeploymentNamespace(), zkSelector, 3, zkPods);
+        kafkaPods = RollingUpdateUtils.waitTillComponentHasRolledAndPodsReady(clusterOperator.getDeploymentNamespace(), kafkaSelector, 3, kafkaPods);
+        eoPods = DeploymentUtils.waitTillDepHasRolled(clusterOperator.getDeploymentNamespace(), KafkaResources.entityOperatorDeploymentName(clusterName), 1, eoPods);
+        // ======== Cluster Operator upgrade ends ========
+
         // ======== Kafka upgrade starts ========
         logPodImages(clusterName);
         changeKafkaAndLogFormatVersion(upgradeData, extensionContext);
         logPodImages(clusterName);
         // ======== Kafka upgrade ends ========
 
-        ClientUtils.waitForClientSuccess(producerName, clusterOperator.getDeploymentNamespace(), messageUpgradeCount);
-        ClientUtils.waitForClientSuccess(consumerName, clusterOperator.getDeploymentNamespace(), messageUpgradeCount);
+        // Send and wait for messages
+        resourceManager.createResource(extensionContext, kafkaBasicClientJob.producerStrimzi(), kafkaBasicClientJob.consumerStrimzi());
```

Comment:
As I mentioned in previous comment, you should just wait for the client success here and not create it